server/app/services/barman-service.js

```diff
@@ -18,12 +18,86 @@ async function getAllBarmen() {
  * Create a Barman.
  *
  * @param newBarman {Barman} partial member
- * @return {Promise<Barman|Errors.ValidationError>} The created barman with its id
+ * @param _embedded {Object} Object containing associations to update, see swagger for more information.
+ * @return {Promise<Barman>} The created barman with its id
  */
-async function createBarman(newBarman) {
+async function createBarman(newBarman, _embedded) {
 
     logger.verbose('Barman service: creating a new barman named %s %s', newBarman.firstName, newBarman.lastName);
-    return await newBarman.save();
+
+    const transaction = await sequelize.transaction();
+    try {
+        await newBarman.save({ transaction });
+
+        // If connection information is changed
+        if (newBarman.connection) {
+            const co = await newBarman.getConnection();
+
+            const coData = {
+                username: newBarman.connection.username,
+                password: hash(newBarman.connection.password)
+            };
+
+            // If there is no connection yet, create one
+            if (!co) {
+                await newBarman.createConnection(cleanObject(coData), { transaction });
+            } else {
+                await co.update(cleanObject(coData), { transaction });
+            }
+        }
+    }catch (err) {
+        logger.warn('Barman service: Error while creating barman', err);
+        await transaction.rollback();
+        throw createServerError('ServerError', 'Error while creating barman');
+    }
+
+    // Associations
+    if (_embedded) {
+        for (const associationKey of Object.keys(_embedded)) {
+            const value = _embedded[associationKey];
+
+            if (associationKey === 'godFather') {
+                const wantedGodFather = await Barman.findById(value);
+
+                if (!wantedGodFather) {
+                    await transaction.rollback();
+                    throw createUserError('UnknownBarman', `Unable to find god father with id ${wantedGodFather}`);
+                }
+
+                await newBarman.setGodFather(wantedGodFather, { transaction });
+
+            } else if (associationKey === 'kommissions') {
+                try {
+                    if (value.add && value.add.length > 0) {
+                        await newBarman.addKommissions(value.add, { transaction });
+                    }
+                    if (value.remove && value.remove.length > 0) {
+                        throw createUserError('No Removed value allowed', 'When creating a barman, impossible to add removed value');
```

Comment:
THe error will be catch by the try and catch

Reply:
And the error name must not contain spaces

Reply:
Ok, but it's not the same kind of error. 
And if i remove the error in the if, what do i have to put in ?